presto-main/src/main/java/io/prestosql/sql/planner/LocalDynamicFiltersCollector.java

```diff
@@ -154,6 +154,12 @@ public synchronized boolean isComplete()
             return futuresLeft == 0;
         }
 
+        @Override
+        public synchronized boolean isAwaitable()
+        {
+            return futuresLeft > 0;
```

Comment:
or `!isBlocked.isDone()` ?

Reply:
`isBlocked` will unblock when DF is narrowed down partially, but DF is still awaitable

Reply:
Yes, but then new `isBlocked` should be created, right?
So, in every moment _current_ value of `isBlocked.isBlocked()` could be used here -- am i right?

Reply:
> Yes, but then new isBlocked should be created, right?

Ah, right. Although `futuresLeft > 0;` seems more explicit